src/assets/CollectiblesController.ts

```diff
@@ -1153,6 +1155,110 @@ export class CollectiblesController extends BaseController<
 
     this.updateNestedCollectibleState(collectibles, ALL_COLLECTIBLES_STATE_KEY);
   }
+
+  /**
+   * Returns the collectible by the address and token id.
+   *
+   * @param address - Hex address of the collectible contract.
+   * @param tokenId - Number that represents the id of the token.
+   * @returns Object containing the Collectible and their position in the array
+   */
+  findCollectibleByAddressAndTokenId(
+    address: string,
+    tokenId: string,
+  ): { collectible: Collectible; index: number } | null {
+    const { allCollectibles } = this.state;
+    const { chainId, selectedAddress } = this.config;
+    const collectibles = allCollectibles[selectedAddress]?.[chainId] || [];
+
+    const index: number = collectibles.findIndex(
+      (collectible) =>
+        collectible.address.toLowerCase() === address.toLowerCase() &&
+        collectible.tokenId === tokenId,
+    );
+
+    if (index === -1) {
+      return null;
+    }
+
+    return { collectible: collectibles[index], index };
+  }
```

Comment:
There are other methods that use the same logic, should we think about refactor them as well?
The methods that I'm talking about are [_updateCollectibleFavoriteStatus_](https://github.com/MetaMask/controllers/blob/05dcaa3084e29e5cb2d4ac284a8c56035e49dfc3/src/assets/CollectiblesController.ts#L1131) and [_checkAndUpdateAllCollectiblesOwnershipStatus_](https://github.com/MetaMask/controllers/blob/05dcaa3084e29e5cb2d4ac284a8c56035e49dfc3/src/assets/CollectiblesController.ts#L1103)
Could be a good solution, instead pulling from this.config, receive the chainId and the selectedAddress from the parameters of the function?

Reply:
Good point. The reason that it's currently ok for those methods to use the `chainId` and `selectedAddress` from config is because they are currently only called to update state for the network and account the user is currently on. By this I mean: the only time a user can change the `favorite` state of a given collectible is when they are on the network and account where they have that collectible. So there isn't as much risk of the state getting out of sync. By contrast updating state related to asynchronous transactions (meaning a user can initiate a transaction and change network/account while the transaction is still pending) requires that we pass along the chainId and selectedAddress through parameters as you suggest above. We honestly probably should refactor those other methods to avoid any potential async issues but that can definitely be in a different PR. Does that make sense?

Reply:
I'm trying to refresh my memory here, so correct me if I'm missing something, but it doesn't look like this method was refactored according to this feedback. Isn't it possible that this method will fail to find the correct NFT if the user changes account/network while the transaction is pending?

Reply:
Can you remind me where you plan to call `updateCollectible` from?

Reply:
Ah! Whooops! Makes sense. Thanks for your patience Tomas!